<launch>

  <arg name="mission_file" default="auction_assign"/>
  <arg name="max_contacts" default="100"/>
  <arg name="plugin_tags" default="REAL, SIL"/>
  <arg name="enable_dyn_reconf" default="false"/>
  <arg name="mission_file_overrides" default="latitude_origin=35.987654321,
                                              longitude_origin=-121.1234567890,
                                              altitude_origin=400.123456789"/>

  <node pkg="scrimmage_ros"
        name="entity_1"
        type="auctioneer"
        output="screen">
    <param name="mission_file" value="$(arg mission_file)" type="str"/>
    <param name="entity_tag" value="entity_1" type="str"/>
    <param name="plugin_tags" type="string" value="$(arg plugin_tags)"/>
    <param name="entity_id" value="1" type="int"/>
    <param name="max_contacts" value="$(arg max_contacts)" type="int"/>
    <param name="loop_rate_hz" type="double" value="10"/>
    <param name="enable_dyn_reconf" type="bool" value="$(arg enable_dyn_reconf)"/>
    <param name="mission_file_overrides" type="string" value="$(arg mission_file_overrides)"/>
  </node>

  <node pkg="scrimmage_ros"
        name="entity_2"
        type="auctioneer"
        output="screen">
    <param name="mission_file" value="$(arg mission_file)" type="str"/>
    <param name="entity_tag" value="entity_2" type="str"/>
    <param name="plugin_tags" type="string" value="$(arg plugin_tags)"/>
    <param name="entity_id" value="2" type="int"/>
    <param name="max_contacts" value="$(arg max_contacts)" type="int"/>
    <param name="loop_rate_hz" type="double" value="10"/>
    <param name="enable_dyn_reconf" type="bool" value="$(arg enable_dyn_reconf)"/>
    <param name="mission_file_overrides" type="string" value="$(arg mission_file_overrides)"/>
  </node>

  <test
    test-name="test_auction"
    pkg="scrimmage_ros"
    type="test_auction">

    <param name="num_bids" value="2" type="int"/>
    <param name="num_starts" value="1" type="int"/>
  </test>

</launch>
